src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInput.java

```diff
@@ -106,17 +112,29 @@ public MongoDbInput( StepMeta stepMeta, StepDataInterface stepDataInterface, int
         if ( meta.getOutputJson() || meta.getMongoFields() == null || meta.getMongoFields().size() == 0 ) {
           String json = JSON.serialize( nextDoc );
           row = RowDataUtil.allocateRowData( data.outputRowMeta.size() );
-          int index = 0;
 
-          row[index++] = json;
+          if ( meta.getExecuteForEachIncomingRow() && m_currentInputRowDrivingQuery != null ) {
+            // Add the incoming columns at start and at the end of the incoming columns add json output.
+            RowMetaInterface inputRowMeta = getInputRowMeta();
+            appendTheIncomingRowsAtStart( row, inputRowMeta );
+            row[inputRowMeta.size()] = json;
```

Comment:
Just code style; we also put spaces before and after square brackets, e.g.: row[ inputRowMeta.size() ].  If the brackets are empty, no spaces are needed, e.g. foo[].